apps/service_providers/llm_service/runnables.py

```diff
@@ -190,80 +142,54 @@ def _chat_is_cancelled(self):
 
         self.last_cancel_check = time.time()
 
-        self.session.chat.refresh_from_db(fields=["metadata"])
-        # temporary mechanism to cancel the chat
-        # TODO: change this to something specific to the current chat message
-        if self.session.chat.metadata.get("cancelled", False):
-            self.cancelled = True
-
+        self.cancelled = self.state.check_cancellation()
         return self.cancelled
 
-    @property
-    def source_material(self):
-        return self.experiment.source_material.material if self.experiment.source_material else ""
-
-    def _input_chain(self) -> Runnable[dict[str, Any], Any]:
-        """Return a langchain runnable that when invoked, will return
-        the fully populated LLM input. This will be used during history compression.
-        """
-        raise NotImplementedError
-
     def _build_chain(self) -> Runnable[dict[str, Any], Any]:
         raise NotImplementedError
 
     @property
     def prompt(self):
         return ChatPromptTemplate.from_messages(
             [
-                ("system", self.experiment.prompt_text),
+                ("system", self.state.get_prompt()),
                 MessagesPlaceholder("history", optional=True),
                 ("human", "{input}"),
             ]
         )
 
     def _populate_memory(self, input: str):
-        # TODO: convert to use BaseChatMessageHistory object
-        model = self.llm_service.get_chat_model(self.experiment.llm, self.experiment.temperature)
-        input_messages = self._input_messages(input)
-        messages = compress_chat_history(
-            self.session.chat, llm=model, max_token_limit=self.experiment.max_token_limit, input_messages=input_messages
-        )
-        self.memory.chat_memory.messages = messages
+        input_messages = self.get_input_messages(input)
```

Comment:
`input_messages` and consequently the chat memory will include the prompt (with source material / participant data etc) if we do this. I'm guessing we don't want this?

Reply:
Apologies, you are correct. I misunderstood what I saw